### ISSUE TITLE
New-AzResourceGroupDeployment fails when deploying virtual network

### ISSUE THREAD

**Issue by segraef** [Azure/azure-powershell#12904]
## Description

New-AzResourceGroupDeployment from Az 4.6.1 fails when deploying virtual network via ARM template.
It works flawless with Az 4.6.0.

![image](https://user-images.githubusercontent.com/19261257/92602308-44c1bc00-f2ae-11ea-801a-0f766402af18.png)

## Steps to reproduce

```powershell
New-AzResourceGroupDeployment -ResourceGroupName snbodnfb -TemplateParameterFile .\Parameters\parameters.json -TemplateFile .\deploy.json -Verbose

```

Once New-AzResourceGroupDeployment from Az 4.6.1 is used it throws following error

## Environment data

```
Name                           Value
----                           -----
PSVersion                      5.1.19041.1
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Module versions

```powershell

```
![image](https://user-images.githubusercontent.com/19261257/92602425-6c188900-f2ae-11ea-8f3b-2540e0c6ae92.png)

## Debug output

```
DEBUG: AzureQoSEvent: CommandName - New-AzResourceGroupDeployment; IsSuccess - False; Duration - 00:00:08.4790201;; Exception -
 System.Exception: 3:10:12 PM - The deployment 'deploy' failed with error(s). Showing 1 out of 1 error(s).
Status Message: Cannot parse the request. (Code: InvalidRequestFormat)
 - Unexpected character encountered while parsing value: [. Path 'properties.addressSpace.addressPrefixes', line 1, position
85. (Code:InvalidJson)
 - JsonToken EndArray is not valid for closing JsonType Object. Path 'properties.addressSpace', line 1, position 87.
(Code:InvalidJson)
 - Infinite loop detected from error handling. Path 'properties.addressSpace', line 1, position 87. (Code:InvalidJson)
 - Additional text encountered after finished reading JSON content: }. Path '', line 1, position 1740. (Code:InvalidJson)
CorrelationId: 0da7a8f3-99f0-45d7-927f-37f138c05bce;
DEBUG: Finish sending metric.
DEBUG: 3:10:13 PM - NewAzureResourceGroupDeploymentCmdlet end processing.
```

## Error output

```
Message        : 3:10:12 PM - The deployment 'deploy' failed with error(s). Showing 1 out of 1 error(s).
                 Status Message: Cannot parse the request. (Code: InvalidRequestFormat)
                  - Unexpected character encountered while parsing value: [. Path 'properties.addressSpace.addressPrefixes',
                 line 1, position 85. (Code:InvalidJson)
                  - JsonToken EndArray is not valid for closing JsonType Object. Path 'properties.addressSpace', line 1,
                 position 87. (Code:InvalidJson)
                  - Infinite loop detected from error handling. Path 'properties.addressSpace', line 1, position 87.
                 (Code:InvalidJson)
                  - Additional text encountered after finished reading JSON content: }. Path '', line 1, position 1740.
                 (Code:InvalidJson)


                 CorrelationId: 0da7a8f3-99f0-45d7-927f-37f138c05bce
StackTrace     :
Exception      : System.Exception
InvocationInfo : {New-AzResourceGroupDeployment}
Line           : New-AzResourceGroupDeployment -ResourceGroupName snbodnfb -TemplateParameterFile .\Parameters\parameters.json
                 -TemplateFile .\deploy.json -Verbose
Position       : At line:1 char:1
                 + New-AzResourceGroupDeployment -ResourceGroupName snbodnfb -TemplatePa ...
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 41
```

**Comment by segraef** [comment]
I can give you the arm template as well if needed, just let me know.

**Comment by dingmeng-xue** [comment]
@segraef , let's check it in email. Actually. Az.Resources has no change between Az.4.6.0 and Az.4.6.1.

**Comment by WilfriedWeidner** [comment]
Hi, I have the same issue during deployment of a VM via automation account runbook.
I try to deploy the VM with my local infrastructure and get the same error.
(az 4.6.1; az.resources 2.5.0)
If I downgrade the module az.resources to 2.4.0 and everything works fine for me!

**Comment by anthony-c-martin** [comment]
@shenglol possible this is related to https://github.com/Azure/azure-powershell/issues/12773?

**Comment by shenglol** [comment]
> @shenglol possible this is related to #12773?
@anthony-c-martin could be the same serialization issue. @segraef do you mind sharing the template and parameters file?

**Comment by shenglol** [comment]
The issue is fixed in Az 4.7.0, which was released yesterday. Closing the issue.
